app/grandchallenge/components/models.py

```diff
@@ -2288,13 +2290,31 @@ def create_civ_for_value(
                 self.add_civ(civ=civ)
                 self.remove_civ(civ=current_civ)
             except ValidationError as e:
+                if created:
+                    civ.delete()
+
                 if new_value in ci.default_field.empty_values:
                     self.remove_civ(civ=current_civ)
                 else:
-                    self.handle_error(
-                        error_message=format_validation_error_message(e),
-                    )
-                    raise e
+                    if hasattr(self, "update_status"):
```

Comment:
It seems like this function might be doing too much, sometimes sending an update status, sometimes sending a notification. I don't really understand when `self` has an `update_status` method or not, and as it is this feels like it would be fragile. Can an `update_status` method be added to the cases where it currently doesn't exist where the notification is sent?

Reply:
It currently does not exist for display sets and archive items. I can add that method there, albeit it would be a bit of a misnomer since these models don't have a `status` to update. I could also call it `handle_error`. For algorithms, all that this `handle_error` function would then do is call the existing `update_status` method that we have for component jobs, fine I guess

Reply:
Yes that sounds good.

Reply:
Ok, that took longer than anticipated. I went back and forth between one method vs, separate model methods; and also wasn't sure whether or not to pass along the upload_session.

I ultimately added a new `handle_civ_validation_error` method that gets attached to all models that inherit from `CIVForObjectMixin`. This method either calls each models own `handle_civ_error` method (doing model specific stuff) or it calls the `update_status` method on the passed upload_session instance. 

I opted to pass along the upload_session so that we can call this new method everywhere in the CIV validation code. The update_status method repeats some of the logic around updating the job status, but also updates the upload session object. Refactoring that method wasn't possible since there are other places that rely on it that are not connected to CIV validation. The alternative would be to call the update_status method directly where that is possible, but then we don't have 1 way of handling errors for CIV validation anymore.....

Reply:
While I was at it, I also changed the job updating so that we now add errors to the `detailed_error_message` rather than overriding it every time. that makes sure that users will see all error messages, not just the latest;
![image](https://github.com/user-attachments/assets/372bfb95-2ec8-48b2-b7c2-23b01f7ab78b)

Reply:
We have to be careful about linking too many things together here, it is quite difficult to follow. Maybe one method of handling errors is not possible, but there should only be one API. Concretely that would mean that we standardise the API of an error handler, a specific error handler for the situation is instantiated when needed (factory method) and then it is called when used. Example of this in the codebase would be the swappable component backends.

For another PR but that representation of error messages could be made nicer, a dictionary to a description list should be straightforward.

Reply:
Ok, I will give that a try.

Reply:
I have pushed a suggestion. Let me know what you think

Reply:
and changing the display to a description list is such a minor change that it doesn't warrant a separate PR, so I have made that change here as well:

![image](https://github.com/user-attachments/assets/b83de56d-c0c7-4461-8706-db50526649e8)

Reply:
I think that the tasks are much cleaner now!